src/Swashbuckle.AspNetCore.SwaggerGen/Annotations/XmlCommentsOperationFilter.cs

```diff
@@ -98,7 +99,13 @@ private void ApplyResponsesXmlToResponses(IDictionary<string, Response> response
             {
                 // Check for a corresponding  API parameter (from ApiExplorer) that's property-bound?
                 var propertyParam = apiDescription.ParameterDescriptions
-                    .Where(p => p.ModelMetadata?.ContainerType != null && p.ModelMetadata?.PropertyName != null)
+                    .Where(p =>
+                    {
+                        var modelMetadata = p.ModelMetadata;
+                        return modelMetadata != null
+                            && modelMetadata.ContainerType != null
+                            && modelMetadata.PropertyName != null;
+                    })
```

Comment:
@xperiandri - do you want to update as discussed and then I'll be happy to merge. Thanks again for the contributions.